File: [BOJ] 백준 풀이/단계별로 풀어보기 (step by step)/#05 함수/1065_한수 (성공).cpp
#include <iostream>

int han(int num)
{
    int i, j, k;
    if (num < 100)
        return 1;
    else
    {
        i = num / 100;
        j = num % 100 / 10;
        k = num % 100 % 10;

        if (i + k == 2 * j && (i < j < k || i > j > k))
            return 1;
        else
            return 0;
    }
}

int main()
{
    int N;
    int temp;
    int count = 0;

    std::cin >> N;

    for (int i = 1; i <= N; i++)
    {
        temp = han(i);
        if (temp == 1)
        {
            count++;
        }
    }

    std::cout << count << "\n";
}

File: 백준 풀이/~23.12.31/#05 함수/1065_한수 (성공).py
def han(num):
    count = 0
    for i in range(1, num+1):
        temp = list(map(int, str(i)))
        if i < 100:
            count += 1
        elif temp[0] - temp[1] == temp[1] - temp[2]:
            count += 1
    
    return count
    
num = int(input())
print(han(num))
